src/lib.rs

```diff
@@ -180,72 +181,42 @@ pub struct IncomingRequests<'a> {
     server: &'a Server
 }
 
-/// Object which allows you to build a server.
-pub struct ServerBuilder {
-    // the address to listen to
-    address: net::SocketAddrV4,
-
-    // number of milliseconds before client timeout
-    client_timeout_ms: u32,
-
-    // maximum number of clients before 503
-    // TODO:
-    //max_clients: usize,
+/// Represents the parameters required to create a server.
+#[derive(Debug, Clone)]
+pub struct ServerConfig<A> where A: ToSocketAddrs {
+    /// The addresses to listen to.
+    pub addr: A,
 }
 
-impl ServerBuilder {
-    /// Creates a new builder.
-    pub fn new() -> ServerBuilder {
-        ServerBuilder {
-            address: net::SocketAddrV4::new(net::Ipv4Addr::new(0, 0, 0, 0), 80),
-            client_timeout_ms: 60 * 1000,
-            //max_clients: { use std::num::Bounded; Bounded::max_value() },
+impl<'a> Default for ServerConfig<(&'a str, u16)> {
+    fn default() -> ServerConfig<(&'a str, u16)> {
```

Comment:
I'm not a huge fan of this default implementation, but I don't really care that much I guess. Could you add a 'doc comment' on this indicating it chooses a random port?